gef-ui/server/controllers/about-facility/validation.js

```diff
@@ -0,0 +1,319 @@
+
+    // if coverEndDate day has validation error
+    if (coverEndDayValidation.error) {
+      coverEndDateValid = false;
+      aboutFacilityErrors.push({
+        errRef: 'coverEndDate',
+        errMsg: 'The day for the cover end date must include 1 or 2 numbers',
+      });
+    }
+    // if coverEndDate month has validation error
+    if (coverEndMonthValidation.error) {
+      coverEndDateValid = false;
+      aboutFacilityErrors.push({
+        errRef: 'coverEndDate',
+        errMsg: 'The month for the cover end date must include 1 or 2 numbers',
+      });
+    }
+    // if coverEndDate year has validation error
+    if (yearValidation.error) {
+      coverEndDateValid = false;
+      aboutFacilityErrors.push({
+        errRef: 'coverEndDate',
+        errMsg: 'The year for the cover end date must include 4 numbers',
+      });
+    }
+  }
+
+  if (coverStartDateIsFullyComplete && coverEndDateIsFullyComplete) {
+    if (coverEndDate < coverStartDate && coverEndDateValid) {
+      aboutFacilityErrors.push({
+        errRef: 'coverEndDate',
+        errMsg: 'Cover end date cannot be before cover start date',
+      });
+    }
+
+    // if coverEndDate is the same as the coverStartDate
+    if (isEqual(coverStartDate, coverEndDate) && coverEndDateValid) {
+      aboutFacilityErrors.push({
+        errRef: 'coverEndDate',
+        errMsg: 'The cover end date must be after the cover start date',
+      });
+    }
+  }
+
+  // validation if should start on submission - no coverStartDate to check against
+  if (shouldCoverStartOnSubmission && coverEndDateIsFullyComplete) {
+    // temporarily set coverStartDate to now
+    const coverStartNow = set(new Date(), {
+      hours: 0,
+      minutes: 0,
+      seconds: 0,
+      milliseconds: 0,
+    });
+
+    if (coverEndDate < coverStartNow && coverEndDateValid) {
+      aboutFacilityErrors.push({
+        errRef: 'coverEndDate',
+        errMsg: 'Cover end date cannot be before cover start date',
+      });
+    }
+
+    if (isEqual(coverStartNow, coverEndDate) && coverEndDateValid) {
+      aboutFacilityErrors.push({
+        errRef: 'coverEndDate',
+        errMsg: 'The cover end date must be after the cover start date',
+      });
+    }
+  }
+
+  // Regex tests to see if value is a number only
+  const digitsRegex = /^[0-9]*$/;
+  if (monthsOfCover) {
+    if (!digitsRegex.test(monthsOfCover)) {
+      aboutFacilityErrors.push({
+        errRef: 'monthsOfCover',
+        errMsg: 'You can only enter numbers',
+      });
+    }
+    if (monthsOfCover > 999) {
+      aboutFacilityErrors.push({
+        errRef: 'monthsOfCover',
+        errMsg: 'You can only enter a maximum of 999 months cover',
+      });
+    }
+  }
+
+  if (isFacilityEndDateEnabled) {
+    if (!facilityEndDateExists && !saveAndReturn) {
+      aboutFacilityErrors.push({
+        errRef: 'facilityEndDateExists',
+        errMsg: 'Select if there is an end date for this facility',
+      });
+    }
+  }
```

Comment:
The majority of this file is just refactoring from `index.js` (commit hash 78276a5d9ba98ae41e290aa62dc3052e2c035af1) - this is the only new validation

Reply:
This is very helpful. Thanks so much :D.

I'm glad you've pulled this into it's own file. I'm very tempted to ask you to pull out the tests regarding errors into a `validation.test` file, how painful would that be?

Reply:
I think quite a bit and not sure the time would be justifiable :/ This function is still doing a lot of stuff